### PR TITLE
Refactor the CSS

### LINKED ISSUE #4009: CSS refactor
A number of visual patterns have coalesced in recent months, so it's a good time to take a closer look at our CSS code and do some cleanup. The goal of this task is to identify and codify current patterns in builder UI, reduce duplication (there's a lot of it) and promote reuse, and upgrade to more recent versions of libraries where applicable (e.g., Neat).

* [x] Replace Bourbon and Neat with browser-native CSS Grid and Flexbox
* [x] Apply uniform styles for page-level components
* [x] Apply uniform styles for lists (navigation lists, toggle lists, action lists, regular ol' lists)
* [x] Apply uniform styles for forms (regular forms, checking-input forms, forms within dialogs)
* [x] Apply uniform styles for dialogs (headers, controls)
* [x] Collapse sidebars on smaller screens
* [x] Code listings (log view, package manifest)
* [x] Handle alternate layouts (Explore, sign-in)
* [x] Retain mobile navigation menus (main nav, sidebar, footer)
* [x] Pass through each view to ensure non-breakage

### PR DESCRIPTION
**Pull request by cnunciato**
This change refactors the CSS in Builder UI to remove redundant and dead code and remove some dependencies (e.g., Bourbon, Neat) in favor of browser-native options like CSS Grid and Flexbox. Also makes Builder UI fully responsive & mobile-first.

Signed-off-by: Christian Nunciato <chris@nunciato.org>

Fixes #4009.

tenor-239361319

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-web/.sass-lint.yml [hunk 1]
```diff
@@ -24,4 +26,6 @@ rules:
   property-sort-order: 0
   quotes:
     - 2
-    - style: double
+    - style: single
```
Comment (raskchanky) [inline components/builder-web/.sass-lint.yml]: tenor-182722408

File: components/builder-web/app/footer/footer.component.html [hunk 2]
```diff
@@ -1,122 +1,65 @@
...
-      <a href="{{config['www_url']}}" title="Habitat Home">
-        <img class="footer--logo habitat" src="../assets/images/habitat-logo-by-chef-white.svg" alt="Habitat by Chef" />
+<div class="footer-component">
+  <footer>
+    <div>
+      <a href="{{config['www_url']}}" title="Habitat.sh">
+        <img src="../assets/images/habitat-logo-by-chef-white.svg" alt="Habitat by Chef">
       </a>
     </div>
-    <div class="footer--links">
```
Comment (raskchanky) [inline components/builder-web/app/footer/footer.component.html]: I'm so stoked all the weird class names are gone.

**Comment by cnunciato** [comment]
Preview: https://cl.ly/3z191S250F1E

File: components/builder-web/app/footer/footer.component.html (hunk 2 above)
Comment (cnunciato) [reply components/builder-web/app/footer/footer.component.html]: There's a system to that -- it's called [BEM](http://getbem.com/) and it's fairly common -- but I'm not a fan, either; I prefer just to use the cascade and to scope by nesting selectors. Some dislike this approach also ("never nest more than three levels deep!"), but I find it maps much more closely to the way I think about what I'm doing, and it hasn't let me down yet. 😄

File: components/builder-web/.sass-lint.yml (hunk 1 above)
Comment (elliott-davis) [reply components/builder-web/.sass-lint.yml]: tenor-117471255

**Comment by cnunciato** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/308043573).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/308043573).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**